### DISCUSSION TITLE
Cloudflare causing Nuxt to fetch data locally instead of SSR and its failing

### DISCUSSION THREAD

**Discussion by gregg-cbs** [nuxt/nuxt#10463] [Questions (Nuxt 2)] [answered]
I would really appreciate some help as I have tried everything and i dont understand why this is happening.

When cloudflare caches mywebsite, it seems Nuxt runs on SSR and then runs again in the browser, but it is running the SSR fetch on the browser and is failing because that is pointing to localhost.

See this video:
https://user-images.githubusercontent.com/57114335/169689211-5129c24d-bad8-4157-b311-3f368be9175e.mov

=========
I would not expect fetch to run in the browser as the fetch is specified to run on SSR and it has a fetchKey.
I disabled javascript caching on cloudflare and I am seeing the same issue still.

**Comment by gregg-cbs** [comment, marked as answer]
Right so i have read a ton of cloudflare and github articles and issues around this matter.

I have switched off HTML and Javascript minification in Cloudflare and this has fixed the issue.

It seems the minification process removes some needed stuff for Nuxt to function. I hope this helps.
https://dev.to/greggcbs/nuxt-and-cloudflare-how-to-get-ssr-working-2-settings-3c48

**Comment by gplusdotgr** [comment]
@gregg-cbs Not sure if you fixed it properly, usually your API endpoint base url needs to be passed through an environment variable in your CI / build; https://nuxtjs.org/docs/configuration-glossary/configuration-env/ 
for example:
```js
  publicRuntimeConfig: {
    baseUrl: process.env.BASE_URL // BASE_URL is passed in your CI/Build depending on what you use
  }
```

then you can use that env variable in your XHR calls:
```js
        const response = await this.$axios.$get(`${this.$config.baseUrl}/api/get-businesses`)
```

this way your dev environment and your production build target different urls as is often the case (the API lives in a different machine/IP address as opposed to local dev where it is usually just `localhost:3000`)

**Reply by gregg-cbs** [reply]
Thanks for your response and this might work, but nuxt shouldnt be making SSR queries from the client. That was the issue I was trying to solve - "why was nuxt making the ssr queries from the client". I answered this below.
